src/ziggurat/producer.clj

```diff
@@ -55,27 +57,77 @@
    :methods  [^{:static true} [send [String String Object Object] java.util.concurrent.Future]
               ^{:static true} [send [String String int Object Object] java.util.concurrent.Future]]))
 
-(defn- producer-properties-from-config [{:keys [bootstrap-servers
-                                                acks
-                                                key-serializer
-                                                value-serializer
-                                                enable-idempotence
-                                                retries-config
-                                                max-in-flight-requests-per-connection]}]
-  (doto (Properties.)
-    (.put ProducerConfig/BOOTSTRAP_SERVERS_CONFIG bootstrap-servers)
-    (.put ProducerConfig/ACKS_CONFIG acks)
-    (.put ProducerConfig/RETRIES_CONFIG (int retries-config))
-    (.put ProducerConfig/ENABLE_IDEMPOTENCE_CONFIG enable-idempotence)
-    (.put ProducerConfig/MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION (int max-in-flight-requests-per-connection))
-    (.put ProducerConfig/KEY_SERIALIZER_CLASS_CONFIG key-serializer)
-    (.put ProducerConfig/VALUE_SERIALIZER_CLASS_CONFIG value-serializer)))
+(defn implements-serializer? [serializer-class]
+  (try
+    (contains? (set (.getInterfaces (Class/forName serializer-class)))
+               (Class/forName "org.apache.kafka.common.serialization.Serializer"))
+    (catch ClassNotFoundException e
+      false)))
+
+(spec/def ::key-serializer-class implements-serializer?)
```

Comment:
This spec library is in the alpha stage of development. Plus, it forces the users to use Clojure 1.10. 

Let's just focus on having all KafkaProducer configs available through this PR. 

Once a decision is taken on removing backward compatibility for Clojure 1.10, we can include this spec library

Reply:
Sure, I can use schema for validation if there are reservations against using spec. However, spec is widely adopted, and I don't see it being it an alpha release as a concern. It's been around for two years and is battle-tested. 

I'll make the change.